src/core/protocol-authorization.ts

```diff
@@ -18,24 +19,25 @@ export class ProtocolAuthorization {
 
   /**
    * Performs protocol-based authorization against the given message.
+   * @param recordsWrite Either the incomingMessage itself if the incoming is a RecordsWrite,
+   *                     or the latest RecordsWrite associated with the recordId being read.
    * @throws {Error} if authorization fails.
    */
   public static async authorize(
     tenant: string,
     incomingMessage: RecordsRead | RecordsWrite,
-    author: string | undefined,
+    recordsWrite: RecordsWrite,
```

Comment:
nit: Maybe there's a better name for what this record represents?

I'm notoriously bad at naming things, so I'm not sure if this is a better suggestion, but maybe something like `rootMessage` or `authorizationTarget`?

Reply:
how bout `newestRecordsWrite`?

Reply:
Since I explain the variable in a comment, I'm gonna leave the name as-is for now. If we decide it's too confusing, we can change it later.